### PR TITLE
feature:modify department service and department razor page

### PR DESCRIPTION
**Pull request by MayueCif**

### CODE REVIEW

File: src/ApiGateways/Masa.Auth.ApiGateways.Caller/Services/Organizations/DepartmentService.cs [hunk 1]
```diff
@@ -1,8 +1,37 @@
-﻿namespace Masa.Auth.ApiGateways.Caller.Services.Organizations;
+﻿using Masa.Auth.Contracts.Admin.Organizations;
+
+namespace Masa.Auth.ApiGateways.Caller.Services.Organizations;
 
 public class DepartmentService : ServiceBase
 {
+    string baseUrl = "api/department";
```
Comment (doddgu) [inline src/ApiGateways/Masa.Auth.ApiGateways.Caller/Services/Organizations/DepartmentService.cs:7]: _baseUrl

File: src/Services/Masa.Auth.Service.Admin/Application/Organizations/CommandHandler.cs [hunk 2]
```diff
@@ -10,29 +10,47 @@ public CommandHandler(IDepartmentRepository departmentRepository)
     }
 
     [EventHandler]
-    public async Task CreateDepartmentAsync(AddDepartmentCommand createDepartmentCommand)
+    public async Task AddDepartmentAsync(AddDepartmentCommand createDepartmentCommand)
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/Application/Organizations/CommandHandler.cs:13]: AddDepartmentCommand createDepartmentCommand?

File: src/Services/Masa.Auth.Service.Admin/Application/Organizations/CommandHandler.cs [hunk 3]
```diff
@@ -10,29 +10,47 @@ public CommandHandler(IDepartmentRepository departmentRepository)
...
     {
-        var parent = await _departmentRepository.FindAsync(createDepartmentCommand.ParentId);
-        var department = new Department(createDepartmentCommand.Name, createDepartmentCommand.Description,
-            parent, createDepartmentCommand.Enabled, createDepartmentCommand.StaffIds.ToArray());
-
-        await _departmentRepository.AddAsync(department);
+        var dto = createDepartmentCommand.AddOrUpdateDepartmentDto;
+        var parent = await _departmentRepository.FindAsync(dto.ParentId);
+        // Add
+        if (dto.Id == Guid.Empty)
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/Application/Organizations/CommandHandler.cs:18]: 1. Abstract a `BaseUpsertDto` or `BaseAddOrUpdateDto`
2. Add `InferredAction` help determine whether to add or update, maybe you need JsonIgnore this property
3. Finally, use switch or if/else process logic

**Review by doddgu — COMMENTED** [review]
@capdiem

File: src/Contracts/Masa.Auth.Contracts.Admin/Infrastructure/Dtos/BaseUpsertDto.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Masa.Auth.Contracts.Admin.Infrastructure.Dtos;
+
+public class BaseUpsertDto<T> where T : IComparable
+{
+    public T Id { get; set; } = default!;
+
+    [JsonIgnore]
+    public bool HasValue => Id.Equals(default(T));
```
Comment (doddgu) [inline src/Contracts/Masa.Auth.Contracts.Admin/Infrastructure/Dtos/BaseUpsertDto.cs:8]: HasValue????

**Review by doddgu — COMMENTED** [review]
LGTM

File: src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Organization/OrgSheet.razor [hunk 5]
```diff
@@ -1,92 +1,30 @@
...
-            <MCardText>
-                <MTextField class="mx-auto text-h5 .neutral-lighten-4--text" HideDetails="@("auto")" style="width:323px;" Placeholder="@T("Enter department name")" TValue="string"></MTextField>
-                <MCascader Items="_departments" Dense class="mt-6"
-                           ItemText="u => u.Name"
-                           ItemValue="u => u.Id"
-                           ItemChildren="u => u.Children">
+<Masa.Blazor.Presets.PDrawer Width=1200 Value=Show ValueChanged=ShowChanged>
+    <MCard Class="mx-auto">
+        <MCardText>
+            <MTextField @bind-Value="@Dto.Name" class="mx-auto text-h5 .neutral-lighten-4--text" HideDetails="@("auto")" style="width:323px;" Placeholder="@T("Enter department name")"></MTextField>
```
Comment (capdiem) [inline src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Organization/OrgSheet.razor:6]: .neutral-lighten-4--text? .?

File: src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Organization/Organization.razor [hunk 6]
```diff
@@ -34,16 +35,16 @@
                                         <MList>
                                             @if (!context.Item.IsRoot)
                                             {
-                                                <MListItem>
+                                                <MListItem OnClick="async ()=> await Copy(context.Item.Id)">
```
Comment (capdiem) [inline src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Organization/Organization.razor:38]: 这里可以省略async await?

File: src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Organization/Organization.razor.cs [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+        new() { Text = "操作", Value = "Action", Sortable = false, Width = 80 }
+    };
+    PaginationDto<StaffDto> _paginationStaffs = new();
+    UpsertDepartmentDto _upsertDepartmentDto = new();
+    CopyDepartmentDto _copyDepartmentDto = new();
+    GetStaffsDto _getStaffsDto = new GetStaffsDto(1, 10, "", Guid.Empty);
+    DepartmentService DepartmentService => AuthCaller.DepartmentService;
+    StaffService StaffService => AuthCaller.StaffService;
+
+
```
Comment (capdiem) [inline src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Organization/Organization.razor.cs:26]: 多了个空行。

**Review by capdiem — CHANGES_REQUESTED** [review]

**Review by capdiem — APPROVED** [review]
LGTM. @doddgu

**Pull request merged**